autotag.go

```diff
@@ -452,7 +463,28 @@ func (r *GitRepo) calcVersion() error {
 	}
 
 	// append optional build metadata
-	if r.buildMetadata != "" {
+	if r.buildNumber {
```

Comment:
nit:  restructure the chained if/else to be more declarative. set a default, and use guards/ifs to override. Avoid the else/if chain if possible.